libs/datamodel/connectors/mongodb-datamodel-connector/src/validations.rs

```diff
@@ -189,3 +189,29 @@ pub(crate) fn unique_cannot_be_defined_to_id_field(index: IndexWalker<'_>, error
         *attr.span(),
     ));
 }
+
+/// A field name cannot contain the `.` character and it cannot start with `$`.
+pub(crate) fn field_name_uses_valid_characters(field: ScalarFieldWalker<'_>, errors: &mut Diagnostics) {
+    let name = match field.mapped_name() {
```

Comment:
We only look into mapped names, a prisma field cannot already start with `$` and cannot have `.` characters in it.